### PR TITLE
watcher: solana ntt watcher

### PR DESCRIPTION
**Pull request by bingyuyap**

### CODE REVIEW

File: watcher/src/watchers/SolanaWatcher.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ export class SolanaWatcher extends Watcher {
     return this.getConnection().getSlot();
   }
 
-  private async findNextValidBlock(
+  protected async findNextValidBlock(
```
Comment (panoel) [inline watcher/src/watchers/SolanaWatcher.ts:56]: Should we grab the version in utils/solant.ts?  They look the same?

File: watcher/src/utils/solana.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  MessageCompiledInstruction,
+  MessageV0,
+  PublicKey,
+  SolanaJSONRPCError,
+  VersionedBlockResponse,
+  VersionedTransactionResponse,
+} from '@solana/web3.js';
+import { decode } from 'bs58';
+
+export const isLegacyMessage = (message: Message | MessageV0): message is Message => {
```
Comment (panoel) [inline watcher/src/utils/solana.ts:14]: Is this file necessary?  Some of the functions in here were taken from other places.  If there are still needed functions, should they go in the common/src/solana.ts file?

File: watcher/src/watchers/__tests__/NTTSolanaWatcher.test.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  expect(transceiverMessageData.ntt_managerPayload.payload.trimmedAmount.amount).toBe(1000000n);
+  expect(transceiverMessageData.ntt_managerPayload.payload.trimmedAmount.decimals).toBe(8);
+  expect(transceiverMessageData.ntt_managerPayload.payload.recipientChain).toBe(1);
+  expect(transceiverMessageData.ntt_managerPayload.payload.sourceToken.toString('hex')).toBe(
+    '0000000000000000000000001d30e78b7c7fbbcef87ae6e97b5389b2e470ca4a'
+  );
+});
+
+// testing two transactions listed above
+test.skip('getNttMessagesForBlock', async () => {
```
Comment (panoel) [inline watcher/src/watchers/__tests__/NTTSolanaWatcher.test.ts:60]: Didn't this work?

Comment (bingyuyap) [reply watcher/src/watchers/__tests__/NTTSolanaWatcher.test.ts:60]: Yes but getNttMessagesForBlock returns the `lastBlockMessage`. In this case, testing it doesn't do much in CI. This test is here more as a entrypoint for myself to run the watcher easily and test everything as a whole.

File: watcher/src/watchers/SolanaWatcher.ts (hunk 1 above)
Comment (bingyuyap) [reply watcher/src/watchers/SolanaWatcher.ts]: Done!

File: watcher/src/utils/solana.ts (hunk 2 above)
Comment (bingyuyap) [reply watcher/src/utils/solana.ts]: I have moved the used functions to `common/src/solana.ts`. Also deleted the unused functions that were meant to modularize some logic in `SolanaWatcher` when we have time to clean up.

**Review by panoel — APPROVED** [review]
